### PR TITLE
fix: set check ethr signer for node clients

### PR DESCRIPTION
**Pull request by Harasz**
### Description

Add missing set for ethr signer for node clients

### Contributor checklist

- [x] Breaking changes - check for any existing interfaces changes that are not backward compatible, removed method etc.
- [x] Documentation - document your code, add comments for method, remember to check if auto generated docs were updated.
- [x] Tests - add new or updated existed test for changes you made.
- [x] Migration guide - add migration guide for every breaking change.
- [x] Configuration correctness - check that any configuration changes are correct ex. default URLs, chain ids, smart contract verification on [Volta explorer](https://volta-explorer.energyweb.org/) or [EWC explorer](https://explorer.energyweb.org/).

### CODE REVIEW

File: src/modules/signer/signer.service.ts [hunk 1]
```diff
@@ -502,4 +505,43 @@ export class SignerService {
       sig
     )}`;
   }
+
+  /**
+   * Set `_isEthSigner` value based on a signed message.
+   * Generates a test message and signs it.
+   */
+  private async _setIsEthrSigner() {
```
Comment (JGiter) [inline src/modules/signer/signer.service.ts:513]: Why not use _calculatePubKeyAndIdentityToken?

Comment (Harasz) [reply src/modules/signer/signer.service.ts:513]: We can, but _calculatePubKeyAndIdentityToken is making a call to RPC asking about block number, so for performance reason i decide to chose this approach

**Review by JGiter — APPROVED** [review]

File: src/modules/signer/signer.service.ts [hunk 2]
```diff
@@ -502,4 +505,43 @@ export class SignerService {
...
+    const encodedHeader = base64url(JSON.stringify(header));
+    const address = this._address;
+    const payload = {
+      address,
+    };
+
+    const encodedPayload = base64url(JSON.stringify(payload));
+    const token = `0x${Buffer.from(
+      `${encodedHeader}.${encodedPayload}`
+    ).toString('hex')}`;
```
Comment (jrhender) [inline src/modules/signer/signer.service.ts:527]: I don't think we this JWT stuff. I think we just need to sign a message and see if we can recover the address. Is there an advantage to the JWT bit?

**Review by jrhender — CHANGES_REQUESTED** [review]

File: src/modules/signer/signer.service.ts (hunk 2 above)
Comment (Harasz) [reply src/modules/signer/signer.service.ts:527]: Yeah, you are right. simple message is enough

**Comment by jrhender** [comment]
The code looks good, but how hard would it be to add tests for this case?
Sorry, I should have asked this in my initial feedback.

**Comment by JGiter** [comment]
> The code looks good, but how hard would it be to add tests for this case?

I think the simplest way to test is to remove code which currently sets isEthrSigner, like here https://github.com/energywebfoundation/iam-client-lib/blob/e61ff2b7363d6b81b0cb8d34d4d60004a5994c2a/e2e/claims.service.e2e.ts#L208

**Comment by Harasz** [comment]
@jrhender tests added

**Review by jrhender — APPROVED** [review]
Looks good, thanks @Harasz

**Pull request merged**